cores/esp8266/WString.h

```diff
@@ -250,27 +250,29 @@ class String {
             uint16_t cap;
             uint16_t len;
         };
-
-        // SSO is handled by checking the last byte of sso_buff.
-        // When not in SSO mode, that byte is set to 0xff, while when in SSO mode it is always 0x00 (so it can serve as the string terminator as well as a flag)
-        // This allows strings up up to 12 (11 + \0 termination) without any extra space.
-        enum { SSOSIZE = sizeof(struct _ptr) + 4 }; // Characters to allocate space for SSO, must be 12 or more
+        // This allows strings up up to 11 (10 + \0 termination) without any extra space.
+        enum { SSOSIZE = sizeof(struct _ptr) + 4 - 1 }; // Characters to allocate space for SSO, must be 12 or more
+        struct _sso {
+            char     buff[SSOSIZE];
+            unsigned len  : 7;
+            unsigned flag : 1;
+        };
         enum { CAPACITY_MAX = 65535 }; // If size of capacity changed, be sure to update this enum
```

Comment:
A bit of a nitpick, but consider defining this value in terms of the type of _ptr. cap,. That way if cap ever changes this changes automatically.
Alternatively, please update the comment that the max means the max value that can be held in cap, which is if type uint16_t.
Why? it took me a bit to re-understand the reasons behind this.

Reply:
"size of capacity" is odd.  Didn't seem that way when I wrote it before.  Will reword.